packages/client/src/forms/configuration/reducer.ts

```diff
@@ -71,6 +71,27 @@ export const formDraftReducer: LoopReducer<
   | Loop<IFormDraftDataState, actions.FormDraftActions> => {
   switch (action.type) {
     case actions.LOAD_DRAFT:
+      return loop(
+        state,
+        Cmd.run(storage.getItem, {
+          args: ['formDraft'],
+          successActionCreator: actions.getOfflineDataSuccess
+        })
+      )
+
+    case actions.GET_OFFLINE_DATA_SUCCESS: {
+      const offlineDataString = action.payload
+      const offlineData: IFormDraftData = JSON.parse(
+        offlineDataString ? offlineDataString : '{}'
+      )
+
+      if (isEmpty(offlineData)) {
+        return loop(state, Cmd.action(actions.fetchDraft()))
+      }
+      return state
```

Comment:
We need to return a new state with the `offlineData` that was loaded from indexedDB

Reply:
Done